analyzers/src/SonarAnalyzer.ShimLayer.CodeGeneration/SyntaxLightupGenerator.cs

```diff
@@ -1209,7 +1209,7 @@ public SyntaxData(in GeneratorExecutionContext context, XDocument document)
             {
                 for (var current = node; current is not null; current = this.TryGetNode(current.BaseName))
                 {
-                    if (current.WrapperName is null)
+                    if (current.ExistingType is not null)
```

Comment:
WrapperName was not working anymore. It was true for `TypeDeclarationSyntax` when we forced it to be included. Therefore the `RecordDeclarationWrapper` base class was set to `BaseTypeDeclaration` instead of `TypeDeclaration`